<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>922</width>
    <height>832</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>2D data Generator</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <widget class="QTabWidget" name="tabWidget_2">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab_generate">
       <attribute name="title">
        <string>Generate Data</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QWidget" name="widget_12" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout_16">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_6">
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="title">
           <string>Config List</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_11">
           <item row="1" column="0">
            <widget class="QPushButton" name="loadConfig">
             <property name="toolTip">
              <string>load config json file</string>
             </property>
             <property name="text">
              <string>Load</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QPushButton" name="delConfig">
             <property name="toolTip">
              <string>delete exist config in list</string>
             </property>
             <property name="text">
              <string>Delete</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="ListWidget" name="configList">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Expanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>91</width>
               <height>16777215</height>
              </size>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QPushButton" name="createConfig">
             <property name="toolTip">
              <string>create new config copy from selected config row</string>
             </property>
             <property name="text">
              <string>New</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="widget_9" native="true">
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QGroupBox" name="groupBox_3">
             <property name="minimumSize">
              <size>
               <width>300</width>
               <height>0</height>
              </size>
             </property>
             <property name="title">
              <string>Seed List</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QWidget" name="src_dir" native="true">
                <layout class="QHBoxLayout" name="horizontalLayout_15">
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QLabel" name="label_19">
                   <property name="text">
                    <string>Seed Dir:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="seedPath">
                   <property name="text">
                    <string>D:\data\2d generate</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="browse_seed_path">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="minimumSize">
                    <size>
                     <width>10</width>
                     <height>10</height>
                    </size>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>20</width>
                     <height>20</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>...</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QLabel" name="seed_status"/>
              </item>
              <item>
               <widget class="MultiDictPropertyWidget" name="itemsListParam" native="true">
                <property name="minimumSize">
                 <size>
                  <width>500</width>
                  <height>500</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>16777215</height>
                 </size>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QWidget" name="widget_3" native="true">
                <layout class="QHBoxLayout" name="horizontalLayout_3">
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QLabel" name="label_stopcondition">
                   <property name="text">
                    <string>Stop Condition</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QComboBox" name="stop_condition"/>
                 </item>
                 <item>
                  <widget class="QLabel" name="label_33">
                   <property name="text">
                    <string>Generate Num</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="SpinBoxWidget" name="generate_num">
                   <property name="focusPolicy">
                    <enum>Qt::StrongFocus</enum>
                   </property>
                   <property name="maximum">
                    <number>9999999</number>
                   </property>
                   <property name="singleStep">
                    <number>1000</number>
                   </property>
                   <property name="value">
                    <number>0</number>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QWidget" name="widget_5" native="true">
                <layout class="QHBoxLayout" name="horizontalLayout_4">
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QPushButton" name="clear_seed_list">
                   <property name="text">
                    <string>Clear Seed List</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <spacer name="verticalSpacer_2">
                <property name="orientation">
                 <enum>Qt::Vertical</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>40</height>
                 </size>
                </property>
               </spacer>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QWidget" name="widget_4" native="true">
             <layout class="QVBoxLayout" name="verticalLayout">
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QGroupBox" name="groupBox">
                <property name="title">
                 <string>Attachment List</string>
                </property>
                <layout class="QGridLayout" name="gridLayout_2">
                 <item row="2" column="0">
                  <widget class="MultiDictPropertyWidget" name="attachmentList" native="true">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="minimumSize">
                    <size>
                     <width>0</width>
                     <height>100</height>
                    </size>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="0">
                  <widget class="QWidget" name="widget_6" native="true">
                   <layout class="QHBoxLayout" name="horizontalLayout_5">
                    <property name="leftMargin">
                     <number>0</number>
                    </property>
                    <property name="topMargin">
                     <number>0</number>
                    </property>
                    <property name="rightMargin">
                     <number>0</number>
                    </property>
                    <property name="bottomMargin">
                     <number>0</number>
                    </property>
                    <item>
                     <widget class="QLabel" name="label">
                      <property name="text">
                       <string>Attach Dir</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="attachment_path"/>
                    </item>
                    <item>
                     <widget class="QPushButton" name="browse_attach_path">
                      <property name="minimumSize">
                       <size>
                        <width>20</width>
                        <height>0</height>
                       </size>
                      </property>
                      <property name="maximumSize">
                       <size>
                        <width>20</width>
                        <height>16777215</height>
                       </size>
                      </property>
                      <property name="text">
                       <string>...</string>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </widget>
                 </item>
                 <item row="3" column="0">
                  <widget class="QPushButton" name="clear_attach_list">
                   <property name="text">
                    <string>Clear Attachment List</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QLabel" name="attach_status"/>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="generate_algo">
                <property name="enabled">
                 <bool>true</bool>
                </property>
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>300</height>
                 </size>
                </property>
                <property name="title">
                 <string>Generate Algorithm</string>
                </property>
                <layout class="QGridLayout" name="gridLayout">
                 <item row="0" column="0">
                  <widget class="DictPropertyListTableWidget" name="property_widget" native="true">
                   <property name="minimumSize">
                    <size>
                     <width>240</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>16777215</width>
                     <height>16777215</height>
                    </size>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBox_10">
                <property name="title">
                 <string>Start Generate</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_3">
                 <item>
                  <widget class="QWidget" name="dest_dir" native="true">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <layout class="QHBoxLayout" name="horizontalLayout_14">
                    <property name="leftMargin">
                     <number>0</number>
                    </property>
                    <property name="topMargin">
                     <number>0</number>
                    </property>
                    <property name="rightMargin">
                     <number>0</number>
                    </property>
                    <property name="bottomMargin">
                     <number>0</number>
                    </property>
                    <item>
                     <widget class="QLabel" name="label_18">
                      <property name="text">
                       <string>Target Dir:</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QLineEdit" name="destPath">
                      <property name="text">
                       <string>D:\data\2d generate-result</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QPushButton" name="browse_target_dir">
                      <property name="sizePolicy">
                       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                        <horstretch>0</horstretch>
                        <verstretch>0</verstretch>
                       </sizepolicy>
                      </property>
                      <property name="minimumSize">
                       <size>
                        <width>10</width>
                        <height>10</height>
                       </size>
                      </property>
                      <property name="maximumSize">
                       <size>
                        <width>20</width>
                        <height>20</height>
                       </size>
                      </property>
                      <property name="text">
                       <string>...</string>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="preview">
                   <property name="text">
                    <string>Preview</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="randomGenerate">
                   <property name="text">
                    <string>Generate</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="randomGenerate_train">
                   <property name="text">
                    <string>generate and train</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="verticalSpacer">
                   <property name="orientation">
                    <enum>Qt::Vertical</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>20</width>
                     <height>40</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_train">
       <attribute name="title">
        <string> Train and Predict</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_12">
        <item row="0" column="0">
         <widget class="TrainAndPredictWidget" name="widget" native="true"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>922</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuasdf">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSave"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionDebug"/>
    <addaction name="actionSingle_Process"/>
    <addaction name="actionRegression_Test"/>
   </widget>
   <addaction name="menuasdf"/>
   <addaction name="menuTools"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="toolTip">
    <string>generate in single processing if debug</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionSave"/>
   <addaction name="actionSave_As"/>
   <addaction name="actionStart_DLServer"/>
   <addaction name="actionDebug"/>
   <addaction name="actionSingle_Process"/>
   <addaction name="actionRegression_Test"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionStart_DLServer">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Start DLServer</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save in target dir</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save As</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionDebug">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Debug</string>
   </property>
   <property name="toolTip">
    <string>generate in single process when debug</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionSingle_Process">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Single_Process</string>
   </property>
   <property name="toolTip">
    <string>gnenerate all data in single process manner.</string>
   </property>
   <property name="visible">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionRegression_Test">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Regression_Test</string>
   </property>
   <property name="toolTip">
    <string>gnenerate all data without random.</string>
   </property>
   <property name="visible">
    <bool>true</bool>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MultiDictPropertyWidget</class>
   <extends>QWidget</extends>
   <header>ui/MultiDictPropertyWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>DictPropertyListTableWidget</class>
   <extends>QWidget</extends>
   <header>ui/DictPropertyListTableWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>TrainAndPredictWidget</class>
   <extends>QWidget</extends>
   <header>ui/TrainAndPredictWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ListWidget</class>
   <extends>QListWidget</extends>
   <header>ui/ListWidget.h</header>
  </customwidget>
  <customwidget>
   <class>SpinBoxWidget</class>
   <extends>QSpinBox</extends>
   <header>ui/SpinBoxWidget.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>configList</tabstop>
  <tabstop>loadConfig</tabstop>
  <tabstop>delConfig</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
